File: packages/typescript/src/territorial/classifyHazard.ts
import { hazardData } from "./data";

export function classifyHazard(text: string): string {
  const lower = text.toLowerCase();
  for (const rule of hazardData.rules) {
    if (rule.keywords.some((keyword) => lower.includes(keyword))) {
      return rule.type;
    }
  }
  return hazardData.default;
}

File: packages/rust/src/territorial/classify_hazard.rs
use crate::data::territorial::{HAZARD_DEFAULT, RULES};

pub fn classify_hazard(text: &str) -> String {
    let lower = text.to_lowercase();
    for (hazard, keywords) in RULES {
        if keywords.iter().any(|k| lower.contains(k)) {
            return hazard.to_string();
        }
    }
    HAZARD_DEFAULT.to_string()
}
